types/test.ts

```diff
@@ -74,62 +74,62 @@ function testSelectorAsCombiner() {
     subSelector,
   );
 
-  // typings:expect-error
+  // $ExpectError
   selector({foo: ''});
 
-  // typings:expect-error
+  // $ExpectError
   const n: number = selector({bar: {foo: ''}});
 
   const s: string = selector({bar: {foo: ''}});
 }
 
 type Component<P> = (props: P) => any;
 
-declare function connect<S, P, R>(selector: ParametricSelector<S, P, R>):
-  (component: Component<P & R>) => Component<P>;
+declare function connect<S, R, P>(selector: Selector<S, R, P[]>):
+  (component: Component<R & S & ([P] extends [never] ? {} : P)>) => Component<P>;
 
 function testConnect() {
   connect(
     createSelector(
       (state: {foo: string}) => state.foo,
-      foo => ({foo}),
+      foo => ({changeUp: foo}),
     )
   )(props => {
-    // typings:expect-error
+    // $ExpectError
     props.bar;
 
-    const foo: string = props.foo;
+    const foo: string = props.changeUp;
   });
 
   const connected = connect(
     createSelector(
       (state: {foo: string}) => state.foo,
-      (state: never, props: {bar: number}) => props.bar,
-      (foo, bar) => ({foo, baz: bar}),
+      (state: {foo: string}, props: {bar: number}) => props.bar,
```

Comment:
`never` is now a proper bottom type and will not assign across a heterogenous collection of selectors.

The state object _should_ always be the same regardless of the selector, so TypeScript is doing us a favor here by forcing users to use a heterogenous type or `any`.